guide/17-geocoding-guides/part6_custom_geocoders.ipynb

```diff
@@ -0,0 +1,953 @@
+{
```

Comment:
<p>As previously mentioned in <a href="https://app.reviewnb.com/Esri/arcgis-python-api/pull/830/part1_what_is_geocoding.ipynb#How-geocoding-works?" target="_blank">Part 1</a>, <code>Geocoders</code> are tools that can find spatial coordinates of addresses, business names, places of interest, and so on, and output points that can be visualized on a map, inserted as stops for a route, or loaded as input for spatial analysis. <code>Geocoders</code> can also be used to generate batch results for a set of addresses, as well as for reverse geocoding, the process of determining the address at a particular x/y location. </p><p>Geocoders can come from many sources, e.g. Esri, Google Maps, and Epi Info 7.2, with various advantages and disadvantages with regard to ease of use, exporting maps, and the ability to geocode addresses outside of the United States. Keep in mind that geocoders are not always precise. Approximate latitudes and longitudes can be found for most addresses, especially in urban areas, but the exact latitude and longitude can vary by a few decimal points depending on which geocoder you use.  </p><p>A <code>GIS</code> includes one or more geocoders. By default it provides access to the <code>ArcGIS World Geocoding Service</code>, and the <code>geocoding</code> module provides a <code>Geocoder</code> class to interact with that service. You can also construct <code>Geocoder</code> objects from your GIS's custom geocoding service items or from the URL of a geocoding service. Further, the list of geocoders registered with the <code>GIS</code> can be queried using <code>get_geocoders()</code>, a method that returns a list of Geocoder instances. </p><p>In the example below, there are more than one registered Geocoders with the <code>GIS</code>, and the first one uses the <code>Esri World Geocoding Service</code> for geocoding:</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/830/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>